File: Creational/AbstractFactory/C++/products/MacButton.cpp
#include <iostream>
#include "MacButton.h"

/**
 * Concrette object of family (realisation).
 */

void MacButton::paint()
{
	std::cout << "Painting button in Mac style." << std::endl;
}

void MacButton::setPosition(const int x, const int y)
{
	std::cout << "Setting button position to (" << x << ", " << y << ") in Mac style." << std::endl;
}

File: Creational/AbstractFactory/Java/products/MacButton.java
package patterns.creational.abstractfactory.products;

import patterns.creational.abstractfactory.interfaces.IButton;

/**
 * Concrette object of family.
 */

public class MacButton implements IButton
{
	/**
	 * @Override
	 * */
	public void paint()
	{
		System.out.println("Painting button in Mac style.");
	}
	
	/**
	 * @Override
	 * */
	public void setPosition(final int x, final int y)
	{
		System.out.println("Setting button position to (" + x + ", " + y + ") in Mac style.");
	}
}
